liblangutil/Token.cpp

```diff
@@ -162,6 +145,32 @@ bool isYulKeyword(string const& _literal)
 
 tuple<Token, unsigned int, unsigned int> fromIdentifierOrKeyword(string const& _literal)
 {
+	// Used for `bytesM`, `uintM`, `intM`, `fixedMxN`, `ufixedMxN`.
+	// M/N must be shortest representation. M can never be 0. N can be zero.
+	auto parseSize = [](string::const_iterator _begin, string::const_iterator _end) -> int
+	{
+		// No number.
+		if (distance(_begin, _end) == 0)
+			return -1;
+
+		// Disallow leading zero.
+		if (distance(_begin, _end) > 1 && *_begin == '0')
+			return -1;
+
+		int ret = 0;
+		for (auto it = _begin; it != _end; it++)
+		{
+			if (*it < '0' || *it > '9')
+				return -1;
+			//  Overflow check. The largest acceptable value is 256 in the callers.
+			if (ret >= 256)
```

Comment:
While this doesn't matter, given when we check this we know that this number will be exceeded, but perhaps a more clear version would have been having this check as the last statement in the loop with `(ret > 256)`. We know that the first iteration is safe as `ret = 0 * 10 + (0..9)` can not overflow.

Reply:
```c++
auto parseSize = +[](string::const_iterator _begin, string::const_iterator _end) -> int
{
  if (_begin == _end) {
    return -1;  // Empty.
  
  } else if (*_begin == '0') {
    return -1;  // Leading zero.
    
  } else if ((_end - _begin) > 3) {
    return -1;  // Too many digits, only need at most three to represent `256`.

  } else {
    int ret = 0;
    for (auto it = _begin; it < _end; ++it) {
      ret = (ret * 10) + (*it - '0');
    }
    return ret;
  }
};
```

Reply:
@pgoodman sorry I missed this.  It still needs a hard check for `ret > 256`, however if that included you feel that this is a useful simplification, please open a PR and I'm sure it can be looked after.